### PR TITLE
[ios][contacts] Fixed bug in updateContactAsync for ios 13

### PR DESCRIPTION
**Pull request by arnaudambro**
Fixed bug in updateContactAsync methods to specify the `keysToFetch` compulsory in ios 13 - because ios13 needs a specific authorization for playing with notes

# Why

[iOS 13 needs a specific authorization for playing with notes](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_contacts_notes). Therefore, if no field is specified to any `expo-contacts` method, the method crashes.
For some methods like [`getContactsAsync`](https://docs.expo.io/versions/latest/sdk/contacts/#getcontactsasync) or [`getContactByIdAsync `](https://docs.expo.io/versions/latest/sdk/contacts/#getcontactbyidasync), it is now compulsory to specify the desired fields. But for the [`updateContactAsync`](https://docs.expo.io/versions/latest/sdk/contacts/#updatecontactasync) method, there is no way to solve the issue in JS, the native code needs to be modified.

# How

`updateContactAsync` calls the `getContactWithId` method, which needs to be passes some specific fields as explained above. We pass all the updated fields.

### CODE REVIEW

**Comment by arnaudambro** [comment]
fixes #5884

**Comment by arnaudambro** [comment]
the error is: `./publish.sh:21: EXPO_CI_ACCOUNT_USERNAME: parameter not set`

no idea how to solve it...

**Comment by EvanBacon** [comment]
Is this not solved in #5601 ?

File: packages/expo-contacts/src/Contacts.ts [hunk 1]
```diff
@@ -307,11 +307,11 @@ export async function addContactAsync(contact: Contact, containerId: string): Pr
   return await ExpoContacts.addContactAsync(contact, containerId);
 }
 
-export async function updateContactAsync(contact: Contact): Promise<string> {
+export async function updateContactAsync(contact: Contact, fields?: FieldType[]): Promise<string> {
   if (!ExpoContacts.updateContactAsync) {
     throw new UnavailabilityError('Contacts', 'updateContactAsync');
   }
-  return await ExpoContacts.updateContactAsync(contact);
+  return await ExpoContacts.updateContactAsync(contact, fields);
```
Comment (EvanBacon) [inline packages/expo-contacts/src/Contacts.ts]: Probably need to add this argument on android as well.

Comment (arnaudambro) [reply packages/expo-contacts/src/Contacts.ts]: @EvanBacon this method is [iOS only](https://docs.expo.io/versions/latest/sdk/contacts/#updatecontactasync)

**Comment by arnaudambro** [comment]
> Is this not solved in #5601 ?

Not in my app with SDK 35...

**Comment by sjchmiela** [comment]
Aye, @EvanBacon's suggestion is correct, this issue has been fixed by https://github.com/expo/expo/commit/18b8f25e800a2c7494819562eaa231128120b6ed. The fact that you still encounter this problem on SDK35 is that this commit hasn't been released yet. It will as a part of SDK36, so the problem will be fixed then!

**Pull request closed without merging**